File: src/main/scala/com/kapilkoju/autopos/party/service/VehicleService.scala
package com.kapilkoju.autopos.party.service

import com.kapilkoju.autopos.party.domain.Vehicle
import com.kapilkoju.autopos.party.repo.VehicleRepo
import org.springframework.data.domain.{Page, Pageable}
import org.springframework.stereotype.Service

trait VehicleService {
  def findVehicles(query: String, pageable: Pageable): Page[Vehicle]
}

@Service
class VehicleServiceImpl(private val vehicleRepo: VehicleRepo)
  extends VehicleService {

  override def findVehicles(query: String, pageable: Pageable) = {
    if (query.isEmpty)
      vehicleRepo.findAll(pageable)
    else
      vehicleRepo.findByNumberIgnoreCaseContainingOrOwnerNameIgnoreCaseContaining(query, query, pageable)
  }
}

File: kotlang/src/main/kotlin/com/kapilkoju/autopos/party/service/VehicleService.kt
package com.kapilkoju.autopos.party.service

import com.kapilkoju.autopos.party.domain.Vehicle
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface VehicleService {
    fun findVehicles(query: String, pageable: Pageable): Page<Vehicle>
}
